_health-care/_js/_components/validated-input.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import _ from 'lodash';
+
+import ErrorMessage from './error-message';
+
+/**
+ * An text input field that can regexp validate an input.
+ *
+ * Validation has the following props.
+ * `pattern` - a regexp that will be used to validate the input.
```

Comment:
Oh that's a good idea! That way we don't need a ton of individual validation functions. Nice :)